File: kezdo-osszeepites-48-legolcsobb-draga-fenyofa/legodf.py
N, ar = [ int(x) for x in input().split() ]
ind = 0
leg = 10000 # 1 <= ar <= 10000
for i in range(N):
    uj = int(input())
    if uj > ar and uj < leg:
        ind = i
        leg = uj
if ind > 0:
    print(ind+1, leg)
else:
    print(-1)

File: kezdo-osszeepites-48-legolcsobb-draga-fenyofa/legodf.cpp
#include <iostream>

using namespace std;

int main()
{
    int N, ar, uj;
    cin >> N >> ar;
    int ind = 0;
    int leg = 10000;
    for(int i=0; i<N; ++i){
        cin >> uj;
        if (uj > ar && uj < leg) {
            ind = i;
            leg = uj;
        }
    }
    if (ind > 0) {
        cout << ind+1 << " " <<leg;
    } else {
        cout << -1;
    }
    return 0;
}
